pkg/querier/tripperware/logicalplan_gen.go

```diff
@@ -0,0 +1,90 @@
+package tripperware
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/thanos-io/promql-engine/logicalplan"
+	"github.com/thanos-io/promql-engine/query"
+	"github.com/weaveworks/common/httpgrpc"
+)
+
+const (
+	stepBatch = 10
+)
+
+func LogicalPlanGenMiddleware() Middleware {
+	return MiddlewareFunc(func(next Handler) Handler {
+		return logicalPlanGen{
+			next: next,
+		}
+	})
+}
+
+type logicalPlanGen struct {
+	next Handler
+}
+
+func (l logicalPlanGen) NewLogicalPlan(qs string, start time.Time, end time.Time, step time.Duration) (*logicalplan.Plan, error) {
+
+	qOpts := query.Options{}
+	if step == 0 {
+		qOpts = query.Options{
+			Start:              start,
+			End:                start,
+			Step:               0,
+			StepsBatch:         stepBatch,
+			LookbackDelta:      0,
+			EnablePerStepStats: false,
+		}
+	} else {
+		qOpts = query.Options{
+			Start:              start,
+			End:                end,
+			Step:               step,
+			StepsBatch:         stepBatch,
+			LookbackDelta:      0,
```

Comment:
nit: Should the lookbackDelta be read from the config?

Reply:
Let's remove `EnablePerStepStats`. As mentioned before I think it is used in execution time only.
We should also set `NoStepSubqueryIntervalFn` for a similar reason. https://github.com/thanos-io/promql-engine/blob/main/query/options.go#L59